src/smk/tool/geomark/tool-geomark.js

```diff
@@ -119,14 +124,35 @@ include.module( 'tool-geomark', [
                     if (!this.defaultHintlineStyle) {
                         this.defaultHintlineStyle = JSON.parse(JSON.stringify(smk.$viewer.map.pm.getGlobalOptions().hintlineStyle));
                     }
-                    smk.$viewer.map.pm.setGlobalOptions({ templineStyle: { color: 'black' }, hintlineStyle: { color: 'green' } });
+                    if (!this.defaultPathOptions) {
+                        this.defaultPathOptions = JSON.parse(JSON.stringify(smk.$viewer.map.pm.getGlobalOptions().pathOptions));
+                    }
+                    smk.$viewer.map.pm.setGlobalOptions({ 
+                        templineStyle: { 
+                            color: '#003366' 
+                        }, 
+                        hintlineStyle: { 
+                            color: '#003366',
+                            fill: false,
+                            dashArray: [5, 5] 
+                        },
+                        pathOptions: {
+                            color: '#003366'
+                        } 
+                    });
+                    smk.$viewer.map.on('pm:create', self.setCurrentDrawingLayer);
                     smk.$viewer.map.pm.enableDraw('Polygon', {
                         continueDrawing: true
                     });
                 }
                 else {
                     smk.$viewer.map.pm.disableDraw();
-                    smk.$viewer.map.pm.setGlobalOptions({ templineStyle: this.defaultTemplineStyle, hintlineStyle: this.defaultHintlineStyle });
+                    smk.$viewer.map.on('pm:create', undefined); // FIXME polygons drawn by Markup still trigger self.setCurrentDrawingLayer 
```

Comment:
I'm looking into this, but if you see an obvious problem please let me know.

Reply:
I don't think this is quite the fix/behaviour we're looking for, but you have the right idea. In this else block, you'll want to unsubscribe from listening for the 'pm:create' event.

I see you are subscribing to the event with: 
`smk.$viewer.map.on('pm:create', self.setCurrentDrawingLayer);`

You can try unsubscribing with:
`smk.$viewer.map.off('pm:create', self.setCurrentDrawingLayer);`

There is a chance this won't work. You might actually need to define setCurrentDrawingLayer like this instead:
`var setCurrentDrawingLayer = function(){ ... }`

Your subscribe and unsubscribe would then look like:
`smk.$viewer.map.on('pm:create', setCurrentDrawingLayer);`
`smk.$viewer.map.off('pm:create', setCurrentDrawingLayer);`

Reply:
This is a test reply